File: dungeons-1.21.11/src/main/java/fishmod/utils/dungeon/FillHelper.java
package fishmod.utils.dungeon;

import fishmod.utils.Constants;
import net.minecraft.client.MinecraftClient;
import net.minecraft.client.network.ClientPlayerEntity;
import net.minecraft.inventory.Inventory;
import net.minecraft.item.ItemStack;

public class FillHelper {

    public static final String ENDER_PEARL = "Ender Pearl";
    public static final String SUPERBOOM_TNT = "Superboom TNT";
    public static final String INFLATABLE_JERRY = "Inflatable Jerry";
    public static final String DECOY = "Decoy";

    /**
     * @param itemName Name of item
     * @param minThreshold an integer that the item count has to be lower than to fill
     * @param maxCount the max count of the item
     * @return 1 if it succeeds else 0 for fail
     */
    public static int fillItem(String itemName, int minThreshold, int maxCount, boolean needAtleastOne) {
        ClientPlayerEntity player = MinecraftClient.getInstance().player;

        if (player == null) return Constants.FAIL;

        Inventory inventory = player.getInventory();

        int currentCount = 0;

        for (int i = 0; i < inventory.size(); i++) {

            ItemStack item = inventory.getStack(i);

            if (item.getName().toString().contains(itemName)) {

                int highestCount = item.getCount();

                if (highestCount > currentCount) {
                    currentCount = highestCount;

                    if (highestCount >= maxCount) return Constants.FAIL;
                }
            }

        }

        if ((needAtleastOne && currentCount == 0) || minThreshold < currentCount) return Constants.FAIL;

        int itemsToGive = maxCount - currentCount;
        player.networkHandler.sendChatCommand("gfs " + itemName + " " + itemsToGive);
        return Constants.SUCCESS;
    }

    public static int fillItem(MinecraftClient client, String itemName, int minThreshold, int maxCount) {
        ClientPlayerEntity player = client.player;

        if (player == null) return Constants.FAIL;

        Inventory inventory = player.getInventory();

        int currentCount = 0;

        for (int i = 0; i < inventory.size(); i++) {

            ItemStack item = inventory.getStack(i);

            if (item.getName().toString().contains(itemName)) {

                int highestCount = item.getCount();

                if (highestCount > currentCount) {
                    currentCount = highestCount;

                    if (highestCount >= maxCount) return Constants.FAIL;
                }
            }

        }

        if ( minThreshold < currentCount) return Constants.FAIL;

        int itemsToGive = maxCount - currentCount;
        player.networkHandler.sendChatCommand("gfs " + itemName + " " + itemsToGive);
        return Constants.SUCCESS;
    }
}

File: dungeons-1.21.11/src/main/kotlin/fishmod/utils/dungeon/FillHelper.kt
package fishmod.utils.dungeon

import fishmod.utils.Constants
import net.minecraft.client.MinecraftClient

object FillHelper {

    const val ENDER_PEARL: String = "Ender Pearl"
    const val SUPERBOOM_TNT: String = "Superboom TNT"
    const val INFLATABLE_JERRY: String = "Inflatable Jerry"
    const val DECOY: String = "Decoy"

    /**
     * @param itemName Name of item
     * @param minThreshold an integer that the item count has to be lower than to fill
     * @param maxCount the max count of the item
     * @return 1 if it succeeds else 0 for fail
     */
    @JvmStatic
    fun fillItem(itemName: String, minThreshold: Int, maxCount: Int, needAtleastOne: Boolean): Int {
        val player = MinecraftClient.getInstance().player ?: return Constants.FAIL

        val inventory = player.inventory

        var currentCount = 0

        for (i in 0 until inventory.size()) {
            val item = inventory.getStack(i)

            if (item.name.toString().contains(itemName)) {
                val highestCount = item.count

                if (highestCount > currentCount) {
                    currentCount = highestCount

                    if (highestCount >= maxCount) return Constants.FAIL
                }
            }
        }

        if ((needAtleastOne && currentCount == 0) || minThreshold < currentCount) return Constants.FAIL

        val itemsToGive = maxCount - currentCount
        player.networkHandler.sendChatCommand("gfs $itemName $itemsToGive")
        return Constants.SUCCESS
    }

    @JvmStatic
    fun fillItem(client: MinecraftClient, itemName: String, minThreshold: Int, maxCount: Int): Int {
        val player = client.player ?: return Constants.FAIL

        val inventory = player.inventory

        var currentCount = 0

        for (i in 0 until inventory.size()) {
            val item = inventory.getStack(i)

            if (item.name.toString().contains(itemName)) {
                val highestCount = item.count

                if (highestCount > currentCount) {
                    currentCount = highestCount

                    if (highestCount >= maxCount) return Constants.FAIL
                }
            }
        }

        if (minThreshold < currentCount) return Constants.FAIL

        val itemsToGive = maxCount - currentCount
        player.networkHandler.sendChatCommand("gfs $itemName $itemsToGive")
        return Constants.SUCCESS
    }
}
